### PR TITLE
Integrate shared branding code

### PR DESCRIPTION
**Pull request by kelvin-chappell**
All the branding logic is now abstracted out into [a dependency](https://github.com/guardian/frontend/compare/kc-integrate-shared?expand=1#diff-0ecdbc5a001d52fb34f5eafb7cd1aaa6R73) so that it can be applied consistently across web and apps.

The branding decisions are now applied directly on the capi model, and then built into the frontend content model.  But this shouldn't have any noticeable effect on the website!
The changes are to:
* all content pages
* all facia pages
* tag pages
* section fronts

The logic for containers and cards is still derived from the frontend content model, so I will change that in the next PR.

/cc @guardian/commercial-dev @guardian/mobile-server-side-staff

### CODE REVIEW

File: common/app/model/PressedPage.scala [hunk 1]
```diff
@@ -105,7 +106,10 @@ case class PressedPage (
 
   val keywordIds: Seq[String] = frontKeywordIds(id)
 
-  override def branding(edition: Edition): Option[Branding] = frontProperties.branding(edition)
-
   def allItems = collections.flatMap(_.curatedPlusBackfillDeduplicated).distinct
+
+  private def hasBrandingOfType(edition: Edition, brandingType: BrandingType): Boolean =
+    metadata.branding(edition).exists(_.brandingType == brandingType)
+  def isSponsored(edition: Edition): Boolean = hasBrandingOfType(edition, Sponsored)
```
Comment (regiskuckaertz) [inline common/app/model/PressedPage.scala]: Is there any advantage in currying `hasBrandingOfType` instead?

File: common/app/model/facia.scala [hunk 2]
```diff
@@ -56,21 +57,27 @@ case class FrontProperties(
...
-  def branding(edition: Edition): Option[Branding] =
-    BrandHunter.findBranding(activeBrandings, edition, publicationDate = None)
 
-  lazy val isAdvertisementFeature: Boolean = {
-    val branding = BrandHunter.findBranding(activeBrandings, defaultEdition, publicationDate = None)
-    branding.exists(_.sponsorshipType == PaidContent)
-  }
+  def branding(edition: Edition): Option[Branding] = EditionBranding.branding(editionBrandings, edition)
+
+  lazy val isAdvertisementFeature: Boolean = branding(defaultEdition).exists(_.brandingType == PaidContent)
```
Comment (regiskuckaertz) [inline common/app/model/facia.scala]: `isPaidContent`?

File: common/app/model/pressedContent.scala [hunk 3]
```diff
@@ -165,9 +165,8 @@ final case class PressedProperties(
   href: Option[String],
   webUrl: Option[String]
 ) {
-  def branding(edition: Edition): Option[Branding] = {
-    maybeContent map (BrandHunter.findContentBranding(_, edition)) getOrElse None
-  }
+  def branding(edition: Edition): Option[Branding] =
+    maybeContent map (_.metadata.branding(edition)) getOrElse None
```
Comment (regiskuckaertz) [inline common/app/model/pressedContent.scala]: do you need the `getOrElse` here?

Comment (regiskuckaertz) [inline common/app/model/pressedContent.scala]: ... also perhaps `flatMap`?

File: common/app/views/support/Commercial.scala [hunk 4]
```diff
@@ -30,10 +31,11 @@ object Commercial {
     s"/guardian-labs$glabsUrlSuffix"
   }
 
-  private def isBrandedContent(page: Page, edition: Edition, sponsorshipType: SponsorshipType): Boolean =
-    page.branding(edition).exists(_.sponsorshipType == sponsorshipType)
+  private def isBrandedContent(page: Page, edition: Edition, brandingType: BrandingType): Boolean =
```
Comment (regiskuckaertz) [inline common/app/views/support/Commercial.scala]: same question wrt currying

**Review by regiskuckaertz — APPROVED** [review]
I just had a few minor questions 👍

File: common/app/model/facia.scala (hunk 2 above)
Comment (kelvin-chappell) [inline common/app/model/facia.scala]: Thanks.  I'll change it.

File: common/app/model/pressedContent.scala (hunk 3 above)
Comment (kelvin-chappell) [inline common/app/model/pressedContent.scala]: No.  Well spotted!

File: common/app/model/PressedPage.scala [hunk 5]
```diff
@@ -27,7 +27,7 @@ object PressedPage {
       "keywords" -> JsString(seoData.webTitle.capitalize),
       "keywordIds" -> JsString(keywordIds.mkString(",")),
       "hasSuperStickyBanner" -> JsBoolean(PersonalInvestmentsCampaign.isRunning(keywordIds)),
-      "isAdvertisementFeature" -> JsBoolean(frontProperties.isAdvertisementFeature)
+      "isAdvertisementFeature" -> JsBoolean(frontProperties.isPaidContent)
```
Comment (rich-nguyen) [inline common/app/model/PressedPage.scala]: can we update all references to `isPaidContent`?

**Comment by rich-nguyen** [comment]
yes, that looks good to me.

**Review by rich-nguyen — APPROVED** [review]

File: common/app/model/PressedPage.scala (hunk 5 above)
Comment (kelvin-chappell) [inline common/app/model/PressedPage.scala]: Yes, but will do it in a separate PR.

File: common/app/model/PressedPage.scala (hunk 1 above)
Comment (kelvin-chappell) [inline common/app/model/PressedPage.scala]: I've cleaned this up a bit.

File: common/app/views/support/Commercial.scala (hunk 4 above)
Comment (kelvin-chappell) [inline common/app/views/support/Commercial.scala]: And I've cleaned this up a bit.
